libraries/Microsoft.Bot.Builder.AI.LUIS/DynamicList.cs

```diff
@@ -35,6 +35,8 @@ public DynamicList()
         /// The lists to append on the extended list entity.
         /// </value>
         [JsonProperty(PropertyName = "list")]
+#pragma warning disable CA2227 // Collection properties should be read only (we can't change this without breaking binary compat)
```

Comment:
>CA2227 [](start = 24, length = 6)

All of these should be both modifiable and settable--we don't want to change this so I disagree with the comments.

Reply:
The recommended approach for these types is to have them initialized with an empty collection (see https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioMessage.cs#L62 for example).

I excluded these checks for backward compat but in the future we shouldn't not expose setters for collections. Newtonsoft serialization works fine with RO lists so that's not an issue.

More info https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1819?view=vs-2019 and https://docs.microsoft.com/en-us/visualstudio/code-quality/CA2227?view=vs-2019